service/popm/popm.go

```diff
@@ -242,23 +244,42 @@ func (m *Miner) bitcoinUTXOs(ctx context.Context, scriptHash []byte) ([]*bfgapi.
 	return buResp.UTXOs, nil
 }
 
-func pickUTXOs(utxos []*bfgapi.BitcoinUTXO, amount int64) ([]*bfgapi.BitcoinUTXO, error) {
-	um := make(map[int]*bfgapi.BitcoinUTXO)
-	for i := range utxos {
-		um[i] = utxos[i]
-	}
+func pickUTXO(utxos []*bfgapi.BitcoinUTXO, amount int64) (*bfgapi.BitcoinUTXO, error) {
+	log.Tracef("pickUTXO")
+	defer log.Debugf("pickUTXO exit")
 
-	// First try to find a single UTXO that equal or exceed the given value.
-	// XXX not random enough
-	for _, utxo := range um {
+	// Filter available UTXOs with value >= amount.
+	var ux []*bfgapi.BitcoinUTXO
+	for i, utxo := range utxos {
 		if utxo.Value >= amount {
-			return []*bfgapi.BitcoinUTXO{utxo}, nil
+			ux = append(ux, utxos[i])
 		}
 	}
 
-	return nil, fmt.Errorf(
-		"address does not have sufficient balance to PoP mine, please send additional funds to continue",
-	)
+	num := len(ux)
+	log.Debugf("Found %d UTXOs (in %d) with value >= %d",
+		num, len(utxos), amount)
+
+	if num == 0 {
+		// There are no available UTXOs with a value >= amount.
+		return nil, errors.New("insufficient funds to PoP mine, " +
+			"please send additional funds to continue mining")
+	}
+
+	utxo := ux[0]
+	if num > 1 {
```

Comment:
nit: I don't think we need this check, and I think we can just get a utxo by `ux[r % num]`